src/cpp/session/modules/SessionRAddins.R

```diff
@@ -14,3 +14,36 @@
 #
 
 assign(".rs.rAddinsEnv", new.env(parent = emptyenv()), envir = .rs.toolsEnv())
+
+.rs.addJsonRpcHandler("prepare_for_addin", function()
+{
+   assign(
+      ".rs.addins.savedShinyResponseFilter",
+      getOption("shiny.http.response.filter"),
+      envir = .rs.toolsEnv())
+   
+   options(shiny.http.response.filter = .rs.addins.shinyResponseFilter)
+})
+
+.rs.addFunction("addins.shinyResponseFilter", function(request, response)
+{
+   # bail if this isn't the root response
+   if (!identical(request$PATH_INFO, "/"))
+      return(response)
+   
+   # remove our old handler
+   options(shiny.http.response.filter = .rs.addins.savedShinyResponseFilter)
+   rm(".rs.addins.savedShinyResponseFilter", envir = .rs.toolsEnv())
+   
+   # ensure text is character rather than raw (not sure if this can happen
+   # but other shiny.http.response.filters do this so best to be safe)
+   if (is.raw(response$content))
+      response$content <- rawToChar(response$content)
+   
+   # inject our CSS into the response
+   css <- '<style type="text/css">select, input { zoom: 1.000001; }</style>'
+   response$content <- sub('<head>', paste('<head>', css, sep = ""), response$content, fixed = TRUE)
```

Comment:
Is `head` always going to be lower case?

Reply:
I'm not sure -- for that reason I should probably use `ignore.case = TRUE` here.